File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> results;
        if(root == nullptr) return results;
        vector<TreeNode*> level;
        vector<int> result;
        int x = 0;
        level.push_back(root);
        int n;
        while(!level.empty()){
            n = level.size();
            for(auto i{0uz}; i < n; ++i){
                result.push_back(level[i] -> val);
                if(level[i] -> left != nullptr) 
                    level.push_back(level[i] -> left);
                if(level[i] -> right != nullptr) 
                    level.push_back(level[i] -> right);
            }

            results.push_back(result);
            for(auto i{0uz}; i < n; ++i){
                level.erase(level.begin());
            }

            result.erase(result.begin(), result.end());
            x++;
        }
        return results;
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []

        memo = []

        level = [root]

        while level:
            memo.append(level[-1].val)
            n = len(level)
            for _ in range(n):
                if level[0].left is not None:
                    level.append(level[0].left)
                if level[0].right is not None:
                    level.append(level[0].right)
                level.pop(0)

        
        return memo
            
